src/QuLDE.jl

```diff
@@ -59,42 +58,41 @@ struct QuLDEUnitParam{W,L,Q, VType <: GeneralMatrixBlock{1}, SType <: GeneralMat
         WS1 = matblock(WS1)
         WS2 = matblock(WS2)
         n = log2i(k+1)
-        new{ComplexF64,L,n, typeof(V), typeof(VS1)}(k,t,C_tilda,D_tilda,N,V,VS1,VS2,WS1,WS2)
+        new{PType, L, n, typeof(V), typeof(VS1)}(k,t,C_tilda,D_tilda,N,V,VS1,VS2,WS1,WS2)
     end
 end
 
-struct QuLDEnonUnitParam{L, W, Q, NL, Nbit, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}, LType <: GeneralMatrixBlock{NL}}
+struct QuLDEnonUnitParam{PType ,L, Q, NL, Nbit, VType <: GeneralMatrixBlock{1}, SType <: GeneralMatrixBlock{Q}, LType <: GeneralMatrixBlock{NL}}
     k::Int # Taylor expansion upto k
     t::L # time
     l::Int # input size of LType block set to 2
-    C_tilda::W #G1 as per the article
-    D_tilda::W #G2 as per the article
-    N::W
+    C_tilda::Complex{PType} #G1 as per the article
+    D_tilda::Complex{PType} #G2 as per the article
+    N::Complex{PType}
     V::VType
     VS1::SType # Q is input size of GMblock
     VS2::SType # Q
     WS1::SType # Q
     WS2::SType # Q
     VT::LType # NL is input size of GMblock
     WT::LType # NL
-    F::Array{GeneralMatrixBlock{Nbit,Nbit,Complex{Float64},Array{Complex{Float64},2}},1} # Nbit is input size
+    F::Array{GeneralMatrixBlock{Nbit,Nbit,Complex{PType},Array{Complex{PType},2}},1} # Nbit is input size
 
-    function QuLDEnonUnitParam(k::Int,t::L,l::Int,prob::QuLDEProblem) where {L}
+    function QuLDEnonUnitParam(k::Int,t::L,l::Int,prob::QuLDEProblem, ::Type{PType} = Float32) where {L,PType}
```

Comment:
In Julia, if there's type for precision etc. we usually define it as the first argument, e.g `rand(Float64, 2, 2)`.